tests/models/test_setup_parsing.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+
+from pdm.models.setup import Setup
+
+
+@pytest.mark.parametrize(
+    "content, result",
+    [
+        (
+            """[metadata]
+name = foo
+version = 0.1.0
+""",
+            Setup("foo", "0.1.0"),
+        ),
+        (
+            """[metadata]
+name = foo
+version = attr:foo.__version__
+""",
+            Setup("foo", "0.0.0"),
+        ),
+        (
+            """[metadata]
+name = foo
+version = 0.1.0
+
+[options]
+python_requires = >=3.6
+install_requires =
+    click
+    requests
+[options.extras_require]
+tui =
+    rich
+""",
+            Setup("foo", "0.1.0", ["click", "requests"], {"tui": ["rich"]}, ">=3.6"),
```

Comment:
Will this work if a version is just `2.0`. Can you please add this scenario too?

Reply:
They are the same case, as long as it is a valid version string.